src/Server.php

```diff
@@ -1696,6 +1692,7 @@ public function addOnlinePlayer(Player $player) : bool{
 	public function removeOnlinePlayer(Player $player) : void{
 		if(isset($this->playerList[$rawUUID = $player->getUniqueId()->getBytes()])){
 			unset($this->playerList[$rawUUID]);
+			unset($this->playerListByUsername[strtolower($player->getName())]);
```

Comment:
This code is not future proof. This could result in a memory leak if the playerList was altered by a plugin (e.g. to hide the player). Future API changes may enable that.